viadot/sources/cloud_for_customers.py

```diff
@@ -5,56 +5,169 @@
 from urllib.parse import urljoin
 from ..config import local_config
 
+import time
+from requests.adapters import HTTPAdapter
+from requests.exceptions import ConnectionError, HTTPError, ReadTimeout, Timeout
+from urllib3.exceptions import ProtocolError
+import logging
+import re
+from requests.packages.urllib3.util.retry import Retry
 
-class CloudForCustomers(Source):
-    """
-    Fetches data from Cloud for Customer.
+url = "https://my341115.crm.ondemand.com/sap/c4c/odata/ana_businessanalytics_analytics.svc/$metadata?entityset=RPZ1F90A299C830A29C6659D3QueryResult"
+
+
+def column_mapper(url=url):
+    credentials = local_config.get("CLOUD_FOR_CUSTOMERS")
+    access = credentials.get("Prod")
+    auth = (access["username"], access["password"])
+    response = requests.get(url, auth=auth)
+    column_mapping = {}
+    for sentence in response.text.split("/>"):
+        result = re.search(r'(?<=Name=")([^"]+).+(sap:label=")([^"]+)+', sentence)
+        if result:
+            key = result.groups(0)[0]
+            val = result.groups(0)[2]
+            column_mapping[key] = val
+    return column_mapping
+
+
+def change_to_meta_url(url):
+    meta_url = ""
+    start = url.split(".svc")[0]
+    ending = url.split("/")[-1]
+    e = ending.split("?")[0]
+    meta_url = start + ".svc/$metadata?entityset=" + e
+    return meta_url
+
+
+def response_to_entity_list(dirty_json, url):
+    metadata_url = change_to_meta_url(url)
+    column_maper_dict = column_mapper(metadata_url)
+    entity_list = []
+    for element in dirty_json["d"]["results"]:
+        new_entity = {}
+        for key, object_of_interest in element.items():
+            if (
+                key != "__metadata"
+                and key != "Photo"
+                and key != ""
+                and key != "Picture"
+            ):
+                # skip the column which contain nested structure
+                if "{" not in str(object_of_interest):
```

Comment:
Are you checking whether the value is a dictionary? Then this should be done like so:
```python
if not isinstance(value, dict):
   ...
```